### DISCUSSION TITLE
Questions in upfront of migrating

### DISCUSSION THREAD

**Discussion by kimpenhaus** [alexjustesen/speedtest-tracker#159] [Q&A] [answered]
Hi Alex,

I am currently using the original version from Henry Whitaker and thought about migrating to your version of the speedtest tracker. Thanks for all the effort you put into bringing it to new levels.

But before switching I was curious about two things:

1. I saw you've changed config to yaml. Is there any auto migration or is it better to start from scratch (maybe because of breaking changes... which leads to my second question)

2. What happened to the health check integration? Looks like it was removed? That is crucial to me unless gotify is integrated. You have any idea when either or will be available?

Thanks,
Marcus

**Comment by alexjustesen** [comment, marked as answer]
Hey Marcus,

Thanks for taking a look at the project! I'll shed some light on your questions below...

1. yaml config was dropped in `v0.1.0`. I'm now using `.env` config which is Laravel's default for anything at the application layer. For settings like notifications and thresholds that's now been moved into the admin panel so you don't have to edit files anymore.
2. Integration for https://healthchecks.io I've rolled into a general topic called "[Application Monitoring](https://docs.speedtest-tracker.dev/getting-started/features)" as I personally use https://ohdear.app and https://thenping.me. I don't have an ETA on those yet as I'm currently working on making sure the container runs as smoothly as possible but they're a requirement prior to tagging a `v1.0.0` release.

Thanks for following along and hope this helps.

**Comment by kimpenhaus** [comment]
Hi Alex,

thanks for the detailed answers and pointing to the Application Monitoring (haven't found that before asking). I will keep an eye on the improvements and once monitoring is supported surely switch.

Again thanks for all the effort you put into it.

Keep well,
Marcus

**Reply by alexdelprete** [reply]
> once monitoring is supported surely switch

I monitor it with Uptime Kuma. The app supports `/ping` endpoint, it answers with `pong`.

Here's the Uptime Kuma config with the appropriate HTTPS Keyword test:

![image](https://user-images.githubusercontent.com/7027842/202877514-2f66d8e1-22b0-4655-bd3a-710cfff5e67c.png)

**Comment by kimpenhaus** [comment]
Good morning Alessandro,

thanks for your suggestion - I am running an instance of [uptime-kuma](https://github.com/louislam/uptime-kuma) in my local network as well, but this monitors (as fa as I can tell) only the general availability. It doesn't give you an idea if:

1. the cron / speediest was executed <--- please correct me if I am wrong
2. what the status of your internet connection is (which could truly be monitored by checking an external url - but is's nice to have in one check)

Cheers,
m.

**Reply by alexdelprete** [reply]
Godde evening Marcus,

> 1. the cron / speediest was executed

as far as I know, `healthchecks.io` doesn't tell you if a speedtest is executed, so I don't understand the objection. These services only tell you if a service is up and running, not what it's doing.

> 2\. what the status of your internet connection is (which could truly be monitored by checking an external url - but is's nice to have in one check)

Same as point 1, but you need to be more precise on what you mean by "status": I monitor my internet connection status constantly in two ways, I have an internal instance of Uptime Kuma that monitors all my homelab containers/services, but I also have an instance on the cloud (fly.io) that monitors 4-5 specific things, one of which is my internet connection from the outside. All this is perfectly achievable with Uptime Kuma without any complexity.

If you want to know the INSIDE status of a process/service, you need much more than healthchecks.io. An app could certainly provide some webhooks that could be managed through Uptime Kuma (I don't think healtchecks.io can do that), but the old project didn't do anything like that, IIRC.

**Comment by kimpenhaus** [comment]
Hey Alessandro,

I am using a self hosted instance of [healthchecks](https://github.com/healthchecks/healthchecks) - this is able to monitor successful executions as well as failures (not only if a ping has or hasn't been sent):

<img width="882" alt="image" src="https://user-images.githubusercontent.com/575109/202929414-e1ffce9b-585d-4882-87bf-f0eb8d970702.png">

The original speedtest-tracker is supporting success/failure (but I bet you know the code better than me)

<img width="567" alt="image" src="https://user-images.githubusercontent.com/575109/202929510-586cb70f-0a1b-433b-a241-11d011b254a6.png">

<img width="372" alt="image" src="https://user-images.githubusercontent.com/575109/202929854-332ec4e6-1dc1-4d35-afd6-0269e3cfcb75.png">

The healthcheck instance  has an Apprise/Gotify integration which will then report the status (if it's up or down). I don't have any remote instance running and don't want to.

I am not in a hurry so I am fine with waiting until that integration is implemented - the current speedtest-tracker is a little outdated and the influxdb v2 integration a little [hacky](https://github.com/henrywhitaker3/Speedtest-Tracker/issues/996) when using the v1 adapter but it is good enough and running...

Thanks.

**Reply by alexdelprete** [reply]
That's a webhook integration, I mentioned it in the last paragraph of my previous post. But that doesn't require healtchecks.io necessarily, you can do the same thing with Uptime Kuma (and other services too) provided that the app supports these hooks.

I was wrong when I said the old project didn't support webhooks, sorry about that. :)

> I don't have any remote instance running and don't want to.

I didn't want anything on the outside too...but then an incident happened, and I found out I also needed a couple of checks from the outside: inside monitoring is not enough to cover all the cases.

Luckily, fly.io allows to host a lightweight container like Uptime Kuma for free, and it's also integrated in my domain, so I have another pair of eyes on the cloud without any extra cost.

So summing it up: I agree with you that this project should provide more status check integrations (webhooks, etc.) and  I'd also love to see apprise integration (through apprise API maybe). Alex is a good developer and is committed on this project, I'm sure he will decide for the best.

**Reply by alexdelprete** [reply]
> and the influxdb v2 integration a little [hacky](https://github.com/henrywhitaker3/Speedtest-Tracker/issues/996) when using the v1 adapter but it is good enough and running...

funny fact: the hack you mentioned (imho: it's an influxdb feature more than a hack), at step 12 it says that the dashboard is based on https://grafana.com/grafana/dashboards/16339, which is made by...:)

![image](https://user-images.githubusercontent.com/7027842/202959959-d64976c7-14b0-4584-8f90-0da871fa597f.png)

I already prepared a grafana dashboard for this new project, but didn't release it yet, still fine tuning things, you can read the discussion with Alex (the dev) here: https://github.com/alexjustesen/speedtest-tracker-docs/issues/4
